src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -143,7 +180,10 @@ CHIP_ERROR OnMessageToSend(TransferSession::OutputEvent & event)
     CHIP_ERROR OnTransferSessionBegin(TransferSession::OutputEvent & event)
     {
         assertChipStackLockedByCurrentThread();
-
+        // Once we receive the BDX init, cancel the BDX Init timeout and start the BDX session
+        if (mSystemLayer) {
+            mSystemLayer->CancelTimer(handleBdxInitReceivedTimeoutExpired, this);
+        }
         VerifyOrReturnError(mFabricIndex.HasValue(), CHIP_ERROR_INCORRECT_STATE);
         VerifyOrReturnError(mNodeId.HasValue(), CHIP_ERROR_INCORRECT_STATE);
```

Comment:
Shouldn't we AbortTransfer if this function returns error (in the caller)?  _That's_ the case we're really trying to abort: when we got an init but we've been reset already.  We're not going to reach that "stale transfer" code below in that case, because we are going to return right here.